block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java

```diff
@@ -728,6 +728,15 @@ private void resetState() {
      * Any cleanup that is needed should be done here.
      */
     private void shutdown() {
+        final long blockInProgress = currentStreamingBlockNumber.get();
+        if (blockInProgress != UNKNOWN_BLOCK_NUMBER) {
+            try {
+                publisherManager.handlerIsEnding(blockInProgress, handlerId);
+                publisherManager.closeBlock(null, handlerId);
+            } catch (final RuntimeException e) {
+                LOGGER.log(WARNING, "Exception during handler ending for handler %d".formatted(handlerId), e);
+            }
+        }
```

Comment:
Instead of two try/catch blocks (always a code smell), we could move this to just before line 743 with the following change.
```suggestion
            // try starts above this line
            final long blockInProgress = currentStreamingBlockNumber.get();
            if (blockInProgress != UNKNOWN_BLOCK_NUMBER) {
                publisherManager.handlerIsEnding(blockInProgress, handlerId);
                publisherManager.closeBlock(null, handlerId);
            }
            resetState(); // Moved from line 741
            // following lines are comments, then removeHandler
```

Reply:
✅ Done! thanks!!! Agreed 💯